src/data/resolver/resolvers.ts

```diff
@@ -1,34 +1,91 @@
+import { INode } from 'components/feeds/feeds-stream-list';
 import { IGraphQLContext } from 'data/graphql-context';
+import { set } from 'lodash';
 
 // TODO: is there simplified way to declare resolver type?
 export const feedResolver = async (
   _: object,
-  __: object,
+  { first }: { first: number },
   context: IGraphQLContext
 ) => {
-  return await context.feeds();
+  const result = await context.feeds.getCachedData(first);
+  return {
+    edges: result.map(wrapToEdge),
+  };
 };
 
 export const articlesResolver = async (
   _: object,
-  __: object,
+  { first }: { first: number },
   context: IGraphQLContext
 ) => {
-  return await context.articles();
+  const result = await context.articles.getCachedData(first);
+  return {
+    edges: result.map(wrapToEdge),
+  };
 };
 
-export const feedsStreamResolver = async (
-  _: object,
-  { after, first }: { after: string; first: number },
-  context: IGraphQLContext
-) => {
-  return context.feedsStream(after, first);
-};
+function wrapToEdge<TNode extends { id: string }>(node: TNode) {
+  return { node, cursor: node.id };
+}
 
-export const feedsStreamEdgesResolver = async (
+export const feedsStreamResolver = async () => ({});
+
+export async function* feedsStreamEdgesResolver(
   _: object,
-  { after, first }: { after: string; first: number },
-  context: IGraphQLContext
-) => {
-  return context.feedsStreamEdges();
-};
+  { first }: { first: number },
+  context: IGraphQLContext,
+  info: any
+) {
+  const result = await context.feeds.getCachedData(first);
+
+  if (!result.length) {
+    await context.feeds.getNetworkData(first);
+    const result = await context.feeds.getCachedData(first);
+    setPackageSize(context, info, result.length);
+    yield* yieldResult(result);
+  } else {
+    setPackageSize(context, info, result.length);
```

Comment:
it looks a bit misleading to me to set max items here when you do not really know it for sure just yet.